tardis/visualization/tools/rpacket_visualization.py

```diff
@@ -0,0 +1,691 @@
+                if math.isclose(current_slope, next_slope, rel_tol=1e-10):
+                    rpacket_interactions.append(0)
+                else:
+                    rpacket_interactions.append(last_interaction_type[step_no])
+
+        return rpacket_x, rpacket_y, rpacket_interactions
+
+    def get_coordinates_multiple_packets(self, r_packet_tracker):
+        """
+        Generates an array of array containing x and y coordinates of multiple packets
+
+        Parameters
+        ----------
+        r_packet_tracker : pandas.core.frame.DataFrame
+            contains the rpacket_tracker_df dataframe with data of only specified no_of_packets
+
+        Returns
+        -------
+        numpy.ndarray
+            array of array containing x coordinates, y coordinates and the interactions for multiple packets
+        """
+
+        # for plotting packets at equal intervals throught the circle, we choose thetas distributed uniformly
+        thetas = np.linspace(0, 2 * math.pi, self.no_of_packets + 1)
+        rpackets_x = []
+        rpackets_y = []
+        rpackets_interactions = []
+        # getting coordinates and interaction arrays for all packets
+        for packet_no in range(self.no_of_packets):
+            (
+                rpacket_x,
+                rpacket_y,
+                rpacket_interactions,
+            ) = self.get_coordinates_with_theta_init(
+                r_packet_tracker.loc[packet_no]["r"],
+                r_packet_tracker.loc[packet_no]["mu"],
+                self.sim.model.time_explosion.value,
+                r_packet_tracker.loc[packet_no]["interaction_type"],
+                thetas[packet_no],
+            )
+            rpackets_x.append(rpacket_x)
+            rpackets_y.append(rpacket_y)
+            rpackets_interactions.append(rpacket_interactions)
+        return (
+            np.array(rpackets_x, dtype="object"),
+            np.array(rpackets_y, dtype="object"),
+            np.array(rpackets_interactions, dtype="object"),
+        )
+
+    def get_equal_array_size(self, rpacket_x, rpacket_y, interactions):
+        """
+        creates the coordinate arrays of different packets of same size. This is done for generating frames in animation.
+
+        Parameters
+        ----------
+        rpacket_x : numpy.ndarray
+            x coordinates of packets
+        rpacket_y : numpy.ndarray
+            y coordinates of packets
+        interactions : numpy.ndarray
+            interaction types of packets
+
+        Returns
+        -------
+        numpy.ndarray
+            normalized x coordinate array
+        numpy.ndarray
+            normalized y coordinate array
+        numpy.ndarray
+            normalized interaction types array
+        int
+            size of the biggest array among different packets
+
+        """
+        rpacket_step_no_array_max_size = max(list(map(len, rpacket_x)))
+        for packet_no in range(len(rpacket_x)):
```

Comment:
It took me a while to realise what you are doing here. Might wanna add an inline comment saying you're making all arrays of length `rpacket_step_no_array_max_size` by repeating last element across the remaining length